pkg/handler/handler.go

```diff
@@ -168,21 +150,38 @@ func (o *OpenAPIService) RegisterOpenAPIVersionedService(servePath string, handl
 						continue
 					}
 
+					contentType := accepts.Type + "/" + accepts.SubType
+					w.Header().Set("Content-Type", contentType)
+
+					o.mutex.Lock()
```

Comment:
It feels a bit weird holding a write lock when serving the OpenAPI. Not blocking but maybe we could refactor this lock into the cache instead of locking/unlocking in the serve method.

Reply:
Me too, initially. But a few things to note:
1. There's nothing we can do if we have two requests coming-in for the data and the data isn't ready. We should let the first one that arrived prepare the content, and the others should wait passively.
2. We need the etag and the data to correspond to the same data, so we shouldn't release the lock in between the two
3. Once the data is ready, the lock will be held for a minimal amount of time, that's fine
4. We don't actually hold the lock while "serving" the data, just while the data is being prepared (if it's not), we can still send data to more than 1 client at a time.
I think the code is fine.